### PR TITLE
Adding colors for V6

### PR DESCRIPTION
**Pull request by malik-habib**
Adding the necessary v6 colors. This is tied to this issue in Scoreboard https://github.com/SpiderStrategies/Scoreboard/issues/54468

### CODE REVIEW

**Review by derekstory — APPROVED** [review]

**Comment by mattsgarlata** [comment]
+1. @nathanbowser  I'll leave it to you to merge in pending your review

**Comment by nathanbowser** [comment]
Are we 100% sure we want color names based on versions of the software? Generally I think versioning is best in the hands of git.

**Comment by mattsgarlata** [comment]
We need to support both version 5 and version 6 colors at the same time. There are a few things to consider here, so let's all chat about this during the next team meeting

**Comment by nathanbowser** [comment]
Sounds good. I think a more maintainable and way for us to stay sane is to cut a new release of scoreboard-colors that will be used for Impact version 6.

npm makes it trivial to do so, you'd have something like this in package.json:

`scoreboard-colors-v5`: `^7.5.0`
`scoreboard-colors-v6` `v8`

This way we can use versioning the way software is usually versioned, rather than versioned variable names.

**Comment by derekstory** [comment]
@nathanbowser Can you remind me what needs done here?

**Comment by nathanbowser** [comment]
We need to figure out what colors are being modified for v6 and what colors are brand new. We don't want to prefix variable names with versions. That's why we have version control.

**Comment by derekstory** [comment]
I'm trying to remember. Is there a plan to prevent SCSS errors when we are in V5 if V6 colors are defined in the Impact stylesheet?

**Comment by mattsgarlata** [comment]
We had a conference call where we all put our heads together on how to do this, and we all came up with a great solution, and now I can't remember what it was. @scottoreilly @jerryorr do either of you remember?

It sucks I didn't write this down, but I wasn't expecting Ahmed to quit and for it to be a month later before we got back around to this.

**Comment by scottoreilly** [comment]
Oh, man, I do not remember. But, if we're going to do another call, let's meet up in the new year so it doesn't have to sit over the break.

**Comment by mattsgarlata** [comment]
I'm pretty sure the solution we came up with involve leaving all the v5 stuff alone and then having a separate branch/release for the v6 stuff.

At one point I think we may have discussed having separate CSS bundles for v5 and v6. Then if you switched between v5 and v6 mode it would switch the CSS bundle being used. That could be accomplished by automatically refreshing the browser after typing `v5()` or `v6()` in the console. The CSS bundles for v5 and v6 can be built in parallel to avoid any slowdown in day-to-day development.

**Comment by mattsgarlata** [comment]
It sounds like Nathan and I remember the same thing, so that must be it!

@nathanbowser I think Derek probably will need your help to set up the separate branch or versions or whatever for the v5 vs v6 work.

**Comment by jerryorr** [comment]
Whenever the appropriate branches are set up for `scoreboard-colors`, etc, let me know and I'll figure out getting the dual CSS files built.

**Comment by mattsgarlata** [comment]
@jerryorr are you sure this will be possible? Would you mind checking? I know less about how the CSS is built, but I'm not clear on how you would include two versions of the same library in npm, or if that's even possible

**Comment by jerryorr** [comment]
I think this is doable. I have it working similarly to what @nathanbowser mentioned earlier. I did this in `package.json`:

```
    "scoreboard-colors": "^7.5.0",
    "scoreboard-colors-v5": "github:SpiderStrategies/scoreboard-colors#jerry-v6-test",
```

And then I build two versions of the css files. I build `style.css` using this import:

```
@import "../node_modules/scoreboard-colors/scoreboard-colors";
```

And I build `style-v5.css` using this:

```
@import "../node_modules/scoreboard-colors-v5/scoreboard-colors";
```

The details are a little hacky, and I need to clean it up to actually be convenient to use, but it works. I haven't dug into the whole "run `v5()` in the console to swap css files" thing yet, but that should be doable in some form. If nothing else, we ought to be able to have a cookie or config setting or something that dynamically picks the right css file to use in the `<style>` tag. I can finish off the proof-of-concept if you want, but it all seems doable to me.

**Comment by mattsgarlata** [comment]
Great! @nathanbowser I think the next step is for you to set up the two different versions or branches for us to use

**Comment by mattsgarlata** [comment]
Also @nathanbowser it would be great if you could tackle this soon (i.e. next work day or two). This is blocking Derek's work on the v6 UI, and that's his top priority right now.

**Comment by derekstory** [comment]
@scottoreilly Trying to pick up with Ahmed left off. He stopped after the reds under "status" in the following screenshot. What about the Charts & Misc section. Are those going in here? And are they replacing the the pallette colors defined here or are a completely different thing? 

Pallette colors:
https://github.com/SpiderStrategies/scoreboard-colors/blob/afae7f37049e07c83e96aa5d088af45f6db73a74/scoreboard-colors.scss#L169

Figma:

Screenshot 2023-12-21 at 12 22 48 PM

**Comment by scottoreilly** [comment]
It should be more clear in the “Updated Color Palate” section of https://github.com/SpiderStrategies/Scoreboard/issues/54468#issuecomment-1753859412, but the general idea is that we’re keeping the score/chart colors the same and replacing the UI colors. I suspect all of that stuff under Charts & Misc won’t be used.

**Comment by derekstory** [comment]
@nathanbowser I think this is ready. I removed the `-v6` naming conventions for the color variables. I didn't remove any of the other colors because I'm not sure if any of that stuff can be removed until I start changing styles in Impact. I'll keep a list of any variables that we can potentially remove once I get going.

**Comment by nathanbowser** [comment]
@derekstory Awesome.

@scottoreilly You know how everything *clicked* for me yesterday when you outlined these new colors in figma? Would you mind grabbing a screenshot of these colors outlined in a rectangle and posting it in this issue/PR? I think that'd be super helpful (at least for me) when looking back at things.

File: scoreboard-colors.scss [hunk 1]
```diff
@@ -28,6 +28,59 @@ $color-night: #0F171D;
...
+$color-orange-700: #e26003;
+$color-orange-750: #bf480d;
+$color-orange-800: #943b0f;
+$color-orange-900: #612405;
+$color-orange-950: #3a1503;
+$color-ai-violet-025: #fbfaff;
+$color-ai-violet-050: #eeebff;
+$color-ai-violet-100: #d8cdfe;
+$color-ai-violet-200: #c4a8f5;
+$color-ai-violet-300: #9f72ee;
```
Comment (mattsgarlata) [inline scoreboard-colors.scss:70]: @scottoreilly @derekstory @nathanbowser for my own understanding, do the designs really use all of these various shades of gray, orange, etc? Or are these just baselines that go into all code automatically generated by Figma?

Comment (derekstory) [reply scoreboard-colors.scss:70]: It seems like they're all intended to be added as new colors but maybe I'm misunderstanding something? Here's the design:

Screenshot 2024-01-03 at 9 38 19 AM

Comment (mattsgarlata) [reply scoreboard-colors.scss:70]: Ah hah. Thanks for the screenshot!

Comment (nathanbowser) [reply scoreboard-colors.scss:70]: Yessss, @derekstory Thanks for the screenshot! Scott showed me that yesterday and everything made sense.

**Comment by nathanbowser** [comment]
@derekstory The other week I think we decided that impact would point to this branch `v6-updated-color-palette` for the new colors, rather than me cutting new releases every time you need a change for the new work. So I'm going to hold off on merging this into master.

**Comment by mattsgarlata** [comment]
OK, great. That makes sense. I'm going to close out this PR because the review here is done.

@derekstory the next time we need changes to the scoreboard-colors project, please submit your PR against the [v6-updated-color-palette](https://github.com/SpiderStrategies/scoreboard-colors/tree/v6-updated-color-palette) branch.

**Pull request closed without merging**